src/domain/entities/Questionnaire/QuestionnaireRules.ts

```diff
@@ -0,0 +1,269 @@
+import { Id } from "../Ref";
+import _ from "../generic/Collection";
+import { Question } from "./QuestionnaireQuestion";
+
+const D2_FUNCTIONS = ["d2:hasValue", "d2:daysBetween", "d2:yearsBetween"];
+const D2_OPERATORS = [
+    ">" as const,
+    ">=" as const,
+    "<" as const,
+    "<=" as const,
+    "==" as const,
+    "!=" as const,
+];
```

Comment:
The variable itself is a red flag: "D2" in a entity (also "d2:hasValue", and so on).. There are a different approaches for this; the most simple -yet we cheat a bit- is to say, ok, let's keep the same idea than DHIS2 has, keep it the expression as a text, and just replace the d2 references: d2:hasValue -> fn:hasValue, for example. A most complex solution would be to build a custom structure (as different as the one in DHIS2 as needed), which the data repo would set. That looks time-costly for a small gain. Evaluate if this merits a question for Jorge. What's sure it's that we don't want references to DHIS2 in the domain.

Reply:
I have mapped functions from 'd2:' to 'fn:' in the repository as per suggestion 1, so that there is no reference of d2 in domain.